shuup/front/apps/product_filter/forms.py

```diff
@@ -0,0 +1,387 @@
+            product_ids += ShopProduct.objects.filter(
+                query_dict['price_ranges'],
+                shop=get_shop(request)
+            ).distinct().values_list("product", flat=True)
+        if 'custom_attributes' in query_dict:
+            product_ids += ProductAttribute.objects.filter(
+                query_dict['custom_attributes'],
+            ).distinct().values_list("product_id", flat=True)
+
+        return product_ids
+
+    def get_basic_attributes_query(self, data_key, data_value):
+        basic_attributes = {}
+
+        if query_key in basic_attributes:
+            basic_attributes[query_key].append(data_value)
+        else:
+            basic_attributes[query_key] = [data_value]
+
+        return basic_attributes
+
+    def get_price_query(self, data_value):
+        price_ranges = ''
+        if " - " in data_value:
+            price_values = data_value.split(" - ")
+            min_value = Decimal(price_values[0])
+            max_value = Decimal(price_values[1])
+            if price_ranges == '':
+                price_ranges = Q(default_price_value__range=(min_value, max_value))
+            else:
+                price_ranges |= Q(default_price_value__range=(min_value, max_value))
+        return price_ranges
+
+    def get_custom_attributes_query(self, data_key, data_value):
+        custom_attributes = ''
+        if custom_attributes == '':
+            custom_attributes = Q(
+                attribute_id=data_key,
+                untranslated_string_value=data_value,
+            )
+        else:
+            custom_attributes |= Q(
+                attribute_id=data_key,
+                untranslated_string_value=data_value,
+            )
+        return custom_attributes
+
+    def get_compiled_attribute_query(self, data):
+        query_dict = {}
+        for key, value in data.items():
+            if value and '__' in key:
+                query_values = data_key.split("__")
+                if query_values[0] in ['width', 'height', 'depth',
+                                       'net_weight', 'gross_weight']:
+                    query_dict.update(
+                        {'basic_attributes': self.get_basic_attributes_query(
+                            query_values[0], query_values[1]
+                        )}
+                    )
+                elif query_values[0] == 'price':
+                    query_dict.update(
+                        {'price_ranges': self.get_price_query(
+                            query_values[1]
+                        )}
+                    )
+                else:
+                    query_dict.update(
+                        {'custom_attributes': self.get_custom_attributes_query(
+                            query_values[0], query_values[1]
+                        )}
+                    )
+
+        return query_dict
+
+    def get_filters(self, request, data):
+        query = {}
+        data.pop("q")
+        data.pop("sort")
```

Comment:
Probably not right place to modify the data. Well with for me this caused key error since the sort was missing from data.